some tests failing in cascade:

```
[est-doc-examples-in-cluster/pods](https://api.ci-op-1tms1tsw-1de64.xxxxxxxxxxxxxxxxxxxxxx:6443/api/v1/namespaces/test-doc-examples-in-cluster/pods)"
--- FAIL: TestDocExamples (0.09s)
    --- FAIL: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9092.yaml (0.00s)
        --- FAIL: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9092.yaml/test-0 (61.21s)
    --- FAIL: TestDocExamples/openshift-monitoring_prometheus-k8s_federate_endpoint.yaml (0.00s)
        --- FAIL: TestDocExamples/openshift-monitoring_prometheus-k8s_federate_endpoint.yaml/test-0 (61.27s)
        --- FAIL: TestDocExamples/openshift-monitoring_prometheus-k8s_federate_endpoint.yaml/test-1 (61.40s)
    --- FAIL: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9091.yaml (0.00s)
        --- FAIL: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9091.yaml/test-1 (61.33s)
        --- FAIL: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9091.yaml/test-0 (61.46s)
    --- FAIL: TestDocExamples/openshift-monitoring_alertmanager-main_service_port_9094.yaml (0.00s)
        --- FAIL: TestDocExamples/openshift-monitoring_alertmanager-main_service_port_9094.yaml/test-1 (61.41s)
        --- FAIL: TestDocExamples/openshift-monitoring_alertmanager-main_service_port_9094.yaml/test-0 (61.66s)
    --- PASS: TestDocExamples/openshift-monitoring_prometheus-k8s_service_port_9091.yaml (0.00s)
        --- PASS: TestDocExamples/openshift-monitoring_prometheus-k8s_service_port_9091.yaml/test-0 (6.00s)
        --- PASS: TestDocExamples/openshift-monitoring_prometheus-k8s_service_port_9091.yaml/test-1 (6.14s)
    --- PASS: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9093.yaml (0.00s)
        --- PASS: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9093.yaml/test-0 (6.19s)
        --- PASS: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9093.yaml/test-2 (6.37s)
        --- PASS: TestDocExamples/openshift-monitoring_thanos-querier_service_port_9093.yaml/test-1 (6.52s)
    --- FAIL: TestDocExamples/openshift-monitoring_alertmanager-main_service_port_9092.yaml (0.00s)
        --- FAIL: TestDocExamples/openshift-monitoring_alertmanager-main_service_port_9092.yaml/test-0 (61.76s)
        --- PASS: TestDocExamples/openshift-monitoring_alertmanager-main_service_port_9092.yaml/test-1 (10.14s)
```

--- a/jsonnet/components/cluster-monitoring-operator.libsonnet
+++ b/jsonnet/components/cluster-monitoring-operator.libsonnet
@@ -233,7 +233,7 @@ function(params) {
       {
         apiGroups: ['config.openshift.io'],
         resources: ['proxies'],
-        verbs: ['get'],
+        verbs: ['get', 'list', 'watch'],
       },
       {
         apiGroups: ['config.openshift.io'],

--- a/manifests/0000_50_cluster-monitoring-operator_02-role.yaml
+++ b/manifests/0000_50_cluster-monitoring-operator_02-role.yaml
@@ -109,6 +109,8 @@ rules:
   - proxies
   verbs:
   - get
+  - list
+  - watch
 - apiGroups:
   - config.openshift.io
   resources:

--- a/pkg/client/client.go
+++ b/pkg/client/client.go
@@ -361,6 +361,16 @@ func (c *Client) ConsoleListWatch() *cache.ListWatch {
 	)
 }
 
+// ProxyListWatch watches for the cluster's proxy configuration resource.
+func (c *Client) ProxyListWatch() *cache.ListWatch {
+	return cache.NewListWatchFromClient(
+		c.oscclient.ConfigV1().RESTClient(),
+		"proxies",
+		"",
+		fields.OneTermEqualSelector("metadata.name", ClusterResourceName),
+	)
+}
+
 // ClusterVersionListWatch watches for the cluster version resource.
 func (c *Client) ClusterVersionListWatch() *cache.ListWatch {
 	return cache.NewListWatchFromClient(

--- a/pkg/operator/operator.go
+++ b/pkg/operator/operator.go
@@ -375,6 +375,19 @@ func New(
 	}
 	o.informers = append(o.informers, informer)
 
+	// Watch the cluster Proxy resource to reconcile when proxy settings change.
+	informer = cache.NewSharedIndexInformer(
+		o.client.ProxyListWatch(),
+		&configv1.Proxy{}, resyncPeriod, cache.Indexers{},
+	)
+	_, err = informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(_, newObj interface{}) { o.handleEvent(newObj) },
+	})
+	if err != nil {
+		return nil, err
+	}
+	o.informers = append(o.informers, informer)
+
 	informer = cache.NewSharedIndexInformer(
 		o.client.ClusterOperatorListWatch("ingress"),
 		&configv1.ClusterOperator{}, resyncPeriod, cache.Indexers{},
@@ -667,6 +680,7 @@ func (o *Operator) handleEvent(obj interface{}) {
 		*configv1.Console,
 		*configv1.ClusterOperator,
 		*configv1.ClusterVersion,
+		*configv1.Proxy,
 		*configv1alpha1.ClusterMonitoring:
 		// Log GroupKind and Name of the obj
 		rtObj := obj.(k8sruntime.Object)

--- a/test/e2e/config_test.go
+++ b/test/e2e/config_test.go
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"strings"
 	"testing"
 	"time"
 
@@ -35,6 +36,7 @@ import (
 	v1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 func TestClusterMonitoringOperatorConfiguration(t *testing.T) {
@@ -1364,4 +1366,62 @@ func assertPrometheusRetentionSize(namespace, name, size string) func(*testing.T
 			t.Fatalf("timed out waiting for retention size configuration: %v", err)
 		}
 	}
+}
+
+// TestProxyConfigWatch verifies that CMO watches the cluster Proxy
+// resource and triggers a reconciliation when it changes.
+// Exact injection correctness is covered by unit tests; here we only
+// check that a Proxy spec change propagates to the Prometheus CR env
+// vars. A Proxy change triggers a full CMO sync so all components are
+// reconciled, we only assert on Prometheus for simplicity.
+//
+// It is still possible that an unrelated event triggers the sync.
+func TestProxyConfigWatch(t *testing.T) {
+	ctx := context.Background()
+	proxies := f.OpenShiftConfigClient.ConfigV1().Proxies()
+
+	proxy, err := proxies.Get(ctx, "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+	originalHTTPProxy := proxy.Spec.HTTPProxy
+
+	httpProxyMarker := fmt.Sprintf("http://cmo-e2e-proxy-%d.invalid:3128", time.Now().UnixNano())
+
+	t.Cleanup(func() {
+		prom, err := f.MonitoringClient.Prometheuses(f.Ns).Get(ctx, "k8s", metav1.GetOptions{})
+		require.NoError(t, err)
+
+		patch := []byte(fmt.Sprintf(`{"spec":{"httpProxy":%q}}`, originalHTTPProxy))
+		_, err = proxies.Patch(ctx, "cluster", types.MergePatchType, patch, metav1.PatchOptions{})
+		require.NoError(t, err)
+
+		// Best effort: wait for the rollout triggered by the proxy
+		// cleanup so the next test doesn't start mid-rollout.
+		if err := f.WaitForPrometheusUpdate(ctx, prom); err != nil {
+			t.Logf("waiting for Prometheus update after proxy cleanup: %v", err)
+		}
+	})
+
+	patch := []byte(fmt.Sprintf(`{"spec":{"httpProxy":%q}}`, httpProxyMarker))
+	_, err = proxies.Patch(ctx, "cluster", types.MergePatchType, patch, metav1.PatchOptions{})
+	require.NoError(t, err)
+
+	require.NoError(t, framework.Poll(5*time.Second, 5*time.Minute, func() error {
+		prom, err := f.MonitoringClient.Prometheuses(f.Ns).Get(ctx, "k8s", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		for _, c := range prom.Spec.Containers {
+			if c.Name == "prometheus" {
+				for _, e := range c.Env {
+					if e.Name == "HTTP_PROXY" {
+						if strings.Contains(e.Value, httpProxyMarker) {
+							return nil
+						}
+						return fmt.Errorf("HTTP_PROXY=%q, want it to contain %q", e.Value, httpProxyMarker)
+					}
+				}
+			}
+		}
+		return fmt.Errorf("HTTP_PROXY not set for prometheus container")
+	}))
 }

